eval_converters/helm/adapter.py

```diff
@@ -124,385 +73,233 @@ def get_main_metric_name(run_path: str, schema_path: str) -> str:
 
 		Returns:
-			EvaluationResult in unified schema format
+			EvaluationLog in unified schema format
 		"""
-		# check if raw_data is a dictionary, JSON string, or file path
-		if isinstance(raw_data, dict):
-			data = raw_data
-		elif isinstance(raw_data, (str, bytes)) and (raw_data.strip().startswith('{') or raw_data.strip().startswith('[')):
-			# It's a JSON string
-			data = json.loads(raw_data)
-		else:
-			# Assume it's a file path
-			with open(raw_data, 'r') as f:
-				data = json.load(f)
 
-		
-		scenario_state_dict = data['scenario_state_dict']
-		run_spec_dict = data['run_spec_dict']
-		scenario_dict = data['scenario_dict']
+		scenario_state_dict = raw_data['scenario_state_dict']
+		run_spec_dict = raw_data['run_spec_dict']
+		scenario_dict = raw_data['scenario_dict']
+		stats = raw_data['stats']
 
-		# Load raw data object into a ScenarioState
 		scenario_state = from_dict(data_class=ScenarioState, data=scenario_state_dict)
 		adapter_spec = scenario_state.adapter_spec
+		request_states = scenario_state.request_states
 
-		# Load raw data object into a RunSpec
 		run_spec = from_dict(data_class=RunSpec, data=run_spec_dict)
 
-		# Construct the EvaluationResult components
-		# 1. Model
-		# 1.1. ModelInfo
-		model_info = ModelInfo(
-			name=adapter_spec.model,
-			family=detect_family(adapter_spec.model),
-		)
-
-		# 1.2. Configuration
-		# HELM does not provide context window size, try loading it from model config, else set to 1
-		try:
-			# try getting context window from model deployment
-			deployment = get_model_deployment(adapter_spec.model_deployment)
-			if deployment and deployment.max_sequence_length is not None:
-				context_window = deployment.max_sequence_length
+		stats: List[Stat] = [from_dict(data_class=Stat, data=stat_info) for stat_info in stats]
 
-			# if not available, try loading it from model config
-			else:
-				config = AutoConfig.from_pretrained(adapter_spec.model)
-
-				priority_fields = [
-					"max_position_embeddings",
-					"n_positions",
-					"seq_len",
-					"seq_length",
-					"n_ctx",
-					"sliding_window"
-				]
-
-				context_window = next((getattr(config, f) for f in priority_fields if hasattr(config, f)), None)
-				if context_window is None:
-					context_window = 1
-		
-		except Exception as e:
-			self.logger.warning(f"Error getting context window: {e}")
-			context_window = 1
+		timestamp = str(min(state.result.request_datetime for state in request_states))
 
-		configuration = Configuration(
-			context_window=context_window,
-		)
+		source_data = SourceData(
+            dataset_name=scenario_dict.get('name'),
+            samples_number=len(request_states),
+            sample_ids=[state.instance.id for state in request_states],
+			additional_details={
+				'scenario_name': run_spec.scenario_spec.class_name,
+				'subject': run_spec.scenario_spec.args.get('subject')
```

Comment:
Here you could just do `**run_spec.scenario_spec.args`. "subject" is specific to MMLU.